spirv_parser.cpp

```diff
@@ -381,13 +381,21 @@ void Parser::parse(const Instruction &instruction)
 		auto mode = static_cast<ExecutionMode>(ops[1]);
 		execution.flags.set(mode);
 
-		if (mode == ExecutionModeLocalSizeId)
+		switch (mode)
 		{
+		case ExecutionModeLocalSizeId:
 			execution.workgroup_size.id_x = ops[2];
 			execution.workgroup_size.id_y = ops[3];
 			execution.workgroup_size.id_z = ops[4];
-		}
+			break;
+
+		case ExecutionModeFPFastMathDefault:
+			execution.fp_fast_math_default_id = ops[3];
```

Comment:
I don't think this is a complete implementation. FPFastMathDefault takes both a type and mode ID to control behavior for specific types (?!), so I don't think this can be correct.

Reply:
Ah. Good point. In theory there could be more than one entry. 

I've modified to track multiple entries as a map.

For MSL compilation, only one set of fast-math flags can be applied at compile time, so I've combined them using accumulative AND'ing, so that only the fast-math flags that apply to all FP types will be applied to the overall compile.